### PR TITLE
Fix recasters with constraints

### PR DESCRIPTION
**Pull request by jrapin**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context / Related issue

Recast optimizers could not work with constraints because they were not told a result in some cases

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the [**CONTRIBUTING**](https://facebookresearch.github.io/nevergrad/contributing.html) document and completed the CLA (see [**CLA**](https://facebookresearch.github.io/nevergrad/contributing.html#contributor-license-agreement-cla)).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

File: nevergrad/functions/__init__.py [hunk 1]
```diff
@@ -5,7 +5,6 @@
 
 from .functionlib import ArtificialFunction as ArtificialFunction
 from .functionlib import FarOptimumFunction as FarOptimumFunction
-from .functionlib import FarOptimumFunction as PBT
```
Comment (jrapin) [inline nevergrad/functions/__init__.py:8]: @teytaud this means that PBT has actually never been used in experiments before

File: nevergrad/optimization/base.py [hunk 2]
```diff
@@ -456,7 +456,7 @@ def ask(self) -> p.Parameter:
                     candidate = self.parametrization.spawn_child()
             if candidate.satisfies_constraints():
                 break  # good to go!
-            if self._penalize_cheap_violations:
+            if self._penalize_cheap_violations or self.no_parallelization:
```
Comment (jrapin) [inline nevergrad/optimization/base.py:459]: with non-parallelizable optimizer, we need to provide a tell

File: nevergrad/functions/__init__.py (hunk 1 above)
Comment (teytaud) [reply nevergrad/functions/__init__.py:8]: we have results so it has been run at some point, but maybe never with master.

File: nevergrad/optimization/base.py (hunk 2 above)
Comment (teytaud) [reply nevergrad/optimization/base.py:459]: waow congrats...

**Review by teytaud — APPROVED** [review]
Congrats, this was a tough one.

**Pull request merged**